### ISSUE TITLE
x86 builds crash immediately

### ISSUE THREAD

**Issue by manganese42** [bromite/bromite#357]
**Bromite version**

Version: `75.0.3770.139` ... `76.0.3809.115`
Arch: `x86`
Android version: 9, 10

**Is the bug reproducible with latest version?**
Yes.

**Can the bug be reproduced with corresponding [Chromium](https://github.com/bromite/chromium/releases) version?**
Yes, but only those provided by Bromite. Google Chrome and other Chromium builds (e.g. https://github.com/wchen342/ungoogled-chromium-android/releases) are not affected.

Given it started happening with `75.0.3770.139`, I suspect the cause is "enable CFI in builds".

**Is the bug a crash?**
Yes. Tombstone/Log: [tombstone_00.txt](https://github.com/bromite/bromite/files/3502698/tombstone_00.txt)

**Describe the bug**
It crashes immediately when opening the app.

**To Reproduce**
1. Install the app (e.g. on Android emulator or a real x86 device)
2. Open the app.
3. Crash.

**Expected behavior**
No crash.

**Screenshots**
N/A

**Comment by csagan5** [comment]
The crash is here, according to the debug dump:
```
Stack Trace:
  RELADDR   FUNCTION                                                                          FILE:LINE
  v------>  PrefValueStore::PrefStoreKeeper::Initialize(PrefValueStore*, PrefStore*, PrefValueStore::PrefStoreType)  ../../components/prefs/pref_value_store.cc:29:18
  02e0140b  PrefValueStore::InitPrefStore(PrefValueStore::PrefStoreType, PrefStore*)          ../../components/prefs/pref_value_store.cc:299:0
```

I am doing some tests to find out what causes it, and whether CFI is related or not.

**Comment by csagan5** [comment]
Found this (unanswered) comment upstream: https://bugs.chromium.org/p/chromium/issues/detail?id=469376#c83

**Comment by csagan5** [comment]
@manganese42 documentation mentions explicitly the `SIGILL` issue here: https://www.chromium.org/developers/testing/control-flow-integrity#TOC-Diagnosing-problems-with-the-CFI-instrumentation

So I might try to diagnose it with `use_cfi_diag=true`, but CFI is likely responsible.

**Comment by csagan5** [comment]
@thestinger I will disable CFI (it was suggested by you some time ago) as it causes instability for the x86 platform and likely that's the reason upstream has not yet enabled it.

I am currently verifying whether that is responsible for this crash or not.

**Comment by thestinger** [comment]
It's enabled upstream for x86_64 desktop targets. It's not enabled upstream for Android on arm64 for performance reasons. It's probably not thoroughly tested for Android on x86. You don't need to enable it for Android on x86 yet if it's problematic there. It's ready for arm64 and tested there, but it substantially increases the binary size as it does for the desktop builds which is likely why they aren't ready to enable it yet.

**Comment by csagan5** [comment]
> You don't need to enable it for Android on x86 yet if it's problematic there.

I suggest you do that for Vanadium; for Bromite I am sticking to having the same build settings and patches for all architectures, so I will follow upstream on this.

**Comment by csagan5** [comment]
Fixed in `76.0.3809.129`.

**Comment by csagan5** [comment]
64-bit ARM builds with CFI are also affected by this since v78: https://bugs.chromium.org/p/chromium/issues/detail?id=977583

Upstream has clearly stated there that they do not ship (nor test) 64-bit versions
